activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -18,6 +18,7 @@ module Util
     TAG_NAME_START_REGEXP = /[^#{TAG_NAME_START_REGEXP_SET}]/
     TAG_NAME_FOLLOWING_REGEXP = /[^#{TAG_NAME_START_REGEXP_SET}\-.0-9\u{B7}\u{0300}-\u{036F}\u{203F}-\u{2040}]/
     TAG_NAME_REPLACEMENT_CHAR = "_"
+    TAG_NAME_COMMON_CASE = /\A[a-z][a-z\-]*\z/
```

Comment:
Does the performance change if we make this regex match the full set of valid values rather than just this common case?

Maybe something like?

``` ruby
TAG_NAME_START_REGEXP_SET = ...
TAG_NAME_FOLLOWING_REGEXP_SET = "#{TAG_NAME_START_REGEXP_SET}\-.0-9\u{B7}\u{0300}-\u{036F}\u{203F}-\u{2040}"
SAFE_XML_TAG_NAME = /\A[#{TAG_NAME_START_REGEXP_SET}][#{TAG_NAME_FOLLOWING_REGEXP_SET}]*\z/
```